spotbugs/src/main/java/edu/umd/cs/findbugs/ba/obl/ObligationAnalysis.java

```diff
@@ -215,6 +215,17 @@ public void edgeTransfer(Edge edge, StateSet fact) throws DataflowAnalysisExcept
                             + edge.getSource().getLastInstruction());
                 }
                 fact.deleteObligation(comparedObligation, edge.getTarget().getLabel());
+
+                // closing a Statement closes the ResultSet
+                Obligation statement = database.getFactory().getObligationByName("java.sql.Statement");
+                if (comparedObligation.equals(statement)) {
+                    Obligation resultSet = database.getFactory().getObligationByName("java.sql.ResultSet");
+                    fact.deleteObligation(resultSet, edge.getTarget().getLabel());
+                    if (DEBUG_NULL_CHECK) {
+                        System.out.println("Deleting " + resultSet.toString() + " on edge from comparison "
```

Comment:
Can this be a proper logging statement?

Reply:
I just copied from line 214 to line 215 [ObligationAnalysis.java ](https://github.com/spotbugs/spotbugs/blob/3b4da42bdb627c0bf97fc97629a563edb880e44b/spotbugs/src/main/java/edu/umd/cs/findbugs/ba/obl/ObligationAnalysis.java#L214-L215). Maybe all `system.out.println()` in this file should be converted to proper logging statement to keep consistency?